pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -20,12 +21,15 @@ import (
 )
 
 const (
-	userDataSecretKey   = "userData"
-	requeueAfterSeconds = 20
-	instanceLinkFmt     = "https://www.googleapis.com/compute/v1/projects/%s/zones/%s/instances/%s"
-	kmsKeyNameFmt       = "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s"
-	machineTypeFmt      = "zones/%s/machineTypes/%s"
-	acceleratorTypeFmt  = "zones/%s/acceleratorTypes/%s"
+	userDataSecretKey            = "userData"
+	requeueAfterSeconds          = 20
+	instanceLinkFmt              = "https://www.googleapis.com/compute/v1/projects/%s/zones/%s/instances/%s"
+	kmsKeyNameFmt                = "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s"
+	machineTypeFmt               = "zones/%s/machineTypes/%s"
+	acceleratorTypeFmt           = "zones/%s/acceleratorTypes/%s"
+	windowsScriptMetadataKey     = "sysprep-specialize-script-ps1"
+	windowsScriptSecretName      = "windows-user-data"
```

Comment:
Ideally, this should be an exported constant in [windows](https://github.com/openshift/machine-api-provider-gcp/pull/14/commits/f22bd895a9d859c1168f9f82c0a47e4e13a9ac8c#diff-359c93fbc08f569c3abf79881f0b426e681c46a5e76765d05e921f16fdb3bf99R13)

Reply:
Does this need to be a constant at all?
I would think you'd just want to use the value of `userDataSecret` in the machine spec.

From [the OCP Windows docs](https://docs.openshift.com/container-platform/4.10/windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.html#windows-machineset-aws_creating-windows-machineset-aws), we expect/require users to use the `windows-user-data` secret.

Reply:
i can update this

Reply:
oh wait, actually, @sebsoto , i think i used the value from the WMCO because what if we have a situation where the user needs to actually specify a `userDataSecret` in addition to whatever the WMCO is creating?

Reply:
> what if we have a situation where the user needs to actually specify a userDataSecret in addition to whatever the WMCO is creating

Currently thats not something we support, but it could be nice to have the flexibility to do that in the future if desired